### PR TITLE
Part 2 of input refactoring

### PR DESCRIPTION
**Pull request by Paxxi**
As the title says, this is a continuation of the refactoring I started. More input handling functions are moved from CApplication and globals have been moved into CInputManager as private members instead.

I've tried to hide implementation details to make calling code cleaner, ifdefs are hidden in inputmanager instead of callers having to be aware of this.

The last commit moving key.h/cpp requires some xcode person to fix the project file I assume, make files are sorted.

### CODE REVIEW

**Comment by mkortstiege** [comment]
Nice. Please grab xcode sync from https://github.com/mkortstiege/xbmc/commit/caeb5c73dfda6773269329861feaeba83a7f935b

**Comment by Paxxi** [comment]
thanks @mkortstiege

**Comment by mkortstiege** [comment]
jenkins build this please.

**Comment by garbear** [comment]
jenkins build this please

File: xbmc/windowing/WinEventsSDL.cpp [hunk 1]
```diff
@@ -346,7 +346,7 @@ bool CWinEventsSDL::MessagePump()
       {
         if (0 == (SDL_GetAppState() & SDL_APPMOUSEFOCUS))
         {
-          g_Mouse.SetActive(false);
+          CInputManager::Get().SetMouseActive((false);
```
Comment (garbear) [inline xbmc/windowing/WinEventsSDL.cpp]: `((`

Comment (Paxxi) [inline xbmc/windowing/WinEventsSDL.cpp]: crap, fixed that once and it managed to sneak back in again

**Comment by Paxxi** [comment]
jenkins build this please

**Comment by garbear** [comment]
missed one

```
/Users/Shared/jenkins/workspace/OSX-64/xbmc/osx/OSXTextInputResponder.mm:31:10: fatal error: 'guilib/Key.h' file not found
#include "guilib/Key.h"
```

should be input/Key.h

**Comment by Paxxi** [comment]
jenkins build this please

**Comment by garbear** [comment]
can you rebase?

**Comment by Paxxi** [comment]
should be ok now @garbear

**Comment by Paxxi** [comment]
jenkins build this please just to be sure

**Comment by garbear** [comment]
thanks! this pr has made the input additions in RetroPlayer much more consolidated

File: xbmc/input/InputManager.cpp [hunk 2]
```diff
@@ -389,10 +430,372 @@ bool CInputManager::ProcessJoystickEvent(int windowId, const std::string& joysti
...
+  return m_RemoteControl.IsInUse();
+#else
+  return false;
+#endif
+}
+
+void CInputManager::EnableRemoteControl()
+{
+#if defined(HAS_LIRC) || defined(HAS_IRSERVERSUITE)
+  m_RemoteControl.setUsed(true);
```
Comment (mkortstiege) [inline xbmc/input/InputManager.cpp]: This looks like a change in behavior and is causing issues on Linux. It's now trying to init on every `ProcessSlow()` call.

Comment (Paxxi) [inline xbmc/input/InputManager.cpp]: in what scenario? with lirc enabled and no remote connected I guess. I may have overlooked something there. What kind of issues re you seeing?

Comment (mkortstiege) [inline xbmc/input/InputManager.cpp]: It tries to init over and over again in case there's no socket to bind to.

**Comment by tamland** [comment]
@Paxxi: `INFO: LIRC Initialize: using: /dev/lircd` and `DEBUG: Failed to connect to LIRC. Retry in 10s.` recently started to get dumped to the log every 500ms or so. Could this PR be the cause of that?

**Comment by mkortstiege** [comment]
@tamland it is. You even commented on the PR that is trying to fix it ;)

**Comment by tamland** [comment]
I, should get some sleep now:)

**Pull request merged**